File: LeetCode_Marathons/LCM4/Easy/Move_Zeroes/a.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        n: int = len(nums)
        collision: int = 0
        for i in range(n):
            if nums[i] != 0:
                nums[i - collision] = nums[i]
                if collision != 0: nums[i] = 0
            else: collision += 1

File: LeetCode_Marathons/LCM4/Easy/Move_Zeroes/a.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int collision = 0;
        for (int i = 0; i <= n - 1; i++) {
            if (nums[i] != 0) {
                nums[i - collision] = nums[i];
                if (collision != 0) {
                    nums[i] = 0;
                }
            } else {
                collision++;
            }
        }
    }
}